src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt

```diff
@@ -38,4 +39,40 @@ class ScheduleRuleService {
       throw BadDataException("Schedule rule has times out of order - ${request.startTime} to ${request.endTime}")
     }
   }
+
+  fun buildNewRules(requestedRules: List<ScheduleRuleRequest>, existingActivity: CourseActivity): List<CourseScheduleRule> {
+    val monthStart = LocalDate.now().with(TemporalAdjusters.firstDayOfMonth())
+    return requestedRules.map {
+      validateRequest(it)
+      CourseScheduleRule(
+        id = findExistingRuleId(it, existingActivity.courseScheduleRules),
+        courseActivity = existingActivity,
+        startTime = LocalDateTime.of(monthStart, it.startTime),
```

Comment:
What's all this start of the month stuff?

Reply:
Nomis models the start and end times as ISO date time (even though it's only interested in the times themselves). Oracle shenanigans methinks.